drafts/navigating_concurrency.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Navigating the Evolving Landscape of Concurrency
+subtitle: From async/await to Structured Concurrency
+domain: software-engineering-corner.hashnode.dev
+tags: concurrency, asynchronous, async, multithreading, , opinion-pieces, programming, developer, learning, general-advice, software-development, programming-tips, software-engineering, computer-science
+cover: https://cdn.hashnode.com/res/hashnode/image/upload/v1721214396092/931726f8-75c0-4f90-b985-2cfdc637c11c.jpeg
+publishAs: tiju
+hideFromHashnodeCommunity: false
+--- 
+
+As described in Demystifying Concurrency: Essential Clarifications, concurrency is evolving. It is clear that programming languages are converging on the async/await Task-based pattern for handling idle time. Even new experimental languages like [Roc](https://www.roc-lang.org/tutorial#the-!-suffix) and [Gleam](https://hexdocs.pm/gleam_otp/gleam/otp/task.html#await) are adopting and improving this pattern.
+
+![async timeline](https://cdn.hashnode.com/res/hashnode/image/upload/v1720084949983/7137b347-2846-458a-8137-d6d04dfbfada.png align="center")
+
+The async/await approach is optimized for the sequential part of asynchronous code execution. Articles like [The Function Colour Myth](https://lukasa.co.uk/2016/07/The_Function_Colour_Myth/) explore the benefits and drawbacks in detail. However, they often overlook the parallel aspect of concurrency.
+
+![](https://cdn.hashnode.com/res/hashnode/image/upload/v1721204475980/829b0131-3f59-44b7-88bc-3229259318b8.png align="center")
```

Comment:
same comment as in your other article about the visualisation. with a few tweaks, this could become a great way to describe concurrency visually.